009-ui-and-admin-server-security.md

```diff
@@ -0,0 +1,84 @@
+
+This proposal discusses amendments to support security with the new [Strimzi UI](https://github.com/strimzi/proposals/pull/6) and [Strimzi Admin Server](https://github.com/strimzi/proposals/pull/9) and outlines how this can be done for SCRAM and OAuth2. Browser based UIs do not traditionally use mutual TLS for authentication and authorization but the Admin server API needs to be considered and this proposal suggests a mechanism that can be used to do that. 
+
+The outlined proposal relies on Kafka to perform authentication or authorization and the Admin Server itself does not.
+In a Kafka environment, this may be acceptable but it is plausible that a UI and Admin Server may wish to communicate with non-Kafka environments in the future like the Kube API.
+In these environments authentication and authorization can again be delegated to the backend system.
+A discussion and agreement is needed whether we wish to implement authentication and authorization on the Admin Server itself.
+
+Proceeding with the delegated model, I have 3 sets of diagrams showing a SCRAM-SHA-512 solution, an OAuth2 solution and a Mutual TLS solution with the last one only being relevant to the Admin Server.
+
+## SCRAM-SHA-512
+![Kafka Client using SCRAM-SHA512](./images/scram-sha512-client.png)
+
+Strimzi supports SCRAM-SHA512 using native Kafka facilities, so the Kafka client is setup with credentials for the user and these credentials are passed to the broker where the broker authenticates against user credentials stored in Zookeeper.
+When the request is authenticated, the Principal is passed to a Kafka Simple ACL Authorizer which determines whether the user is authorized to access the resource.
+Access is defined through ACLs and are stored on Zookeeper.
+
+![Strimzi UI using SCRAM-SHA512](./images/scram-sha512-ui.png)
+
+This Kafka client model translates to a UI and Admin Server in a similar way.
+When the user loads up the UI in a browser, they are presented with a dialogue box requiring them to enter a userId and password.
+The userId and password they enter are pre-defined to Kafka by the creation of a Kafka user through the `User Operator` or the Kafka `bin/kafka-configs.sh ` command line tool.
+Once the userId and password have been entered, the UI validates the credentials on Zookeeper using `passport.js`.
+The result that is returned will be marked as valid or invalid.
+If the credentials are valid, they are stored in the session data associated with that browser session.
+When the UI requires data from the Admin Server, the credentials are recovered from the session data and placed as an auth header on the request.
```

Comment:
in which "form" are they set in the auth header? AFAIK in general it's just "user:password" base64 encoded ... should we have TLS encryption between Strimzi UI and admin server?